data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala

```diff
@@ -49,14 +53,75 @@ object StateAdminReportJob extends optional.Application with IJob with StateAdmi
         JobContext.parallelization = 10
         implicit val sparkSession: SparkSession = openSparkSession(jobConfig);
         implicit val frameworkContext = getReportingFrameworkContext();
+        DecryptUtil.initialise();
         execute(jobConfig)
         closeSparkSession()
     }
-
+    
     private def execute(config: JobConfig)(implicit sparkSession: SparkSession, fc: FrameworkContext) = {
 
-        generateReport();
+        /*generateReport();*/
         JobLogger.end("StateAdminReportJob completed successfully!", "SUCCESS", Option(Map("config" -> config, "model" -> name)))
+        
+        generateExternalIdReport();
+        JobLogger.end("ExternalIdReportJob completed successfully!", "SUCCESS", Option(Map("config" -> config, "model" -> name)))
+    }
+    
+    def generateExternalIdReport() (implicit sparkSession: SparkSession, fc: FrameworkContext) = {
+        import sparkSession.implicits._
+    
+        val container = AppConf.getConfig("cloud.container.reports")
+        val objectKey = AppConf.getConfig("admin.metrics.cloud.objectKey")
+        val storageConfig = getStorageConfig(container, objectKey);
+        val usrExternalIdentityEncoder = Encoders.product[UsrExternalIdentity].schema
+        val userExternalDataDF = loadData(sparkSession, Map("table" -> "usr_external_identity", "keyspace" -> sunbirdKeyspace), Some(usrExternalIdentityEncoder)).
+            filter(col(colName = "idtype").isin("declared-email", "declared-phone", "declared-school-name", "declared-school-udise-code", "declared-ext-id", "declared-state", "declared-district")).as[UsrExternalIdentity]
+        val locationDF = locationData();
+        val userExternalOriginalDataDF = userExternalDataDF.groupBy("userid", "provider").pivot("idtype").agg(first("originalexternalid").alias("originalexternalid"))
```

Comment:
pls put a comment why origExternalId is being looked up here.

Reply:
comments are added across the code